src/lib/optimizer/strategy/between_composition_rule.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "abstract_rule.hpp"
+#include "expression/expression_functional.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+class PredicateNode;
+
+/**
+ * The ColumnBoundaryType defines whether a value represents a boundary for a column or not
+ * and if it is a boundary it also defines which kind of boundary it is including the inclusive and exclusive property.
+**/
+enum ColumnBoundaryType {
```

Comment:
pls make this an enum class and remove it from the opossum namespace, instead make it a member type of BetweenCompositionRule